### PR TITLE
Fix typha gob encoding.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

The encoding must use the pre-monorepo package names for back compatibility.

Add a test to prevent further regressions.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: typha/pkg/syncproto/sync_proto_test.go
Comment (neiljerram) [inline typha/pkg/syncproto/sync_proto_test.go:80]: Test that `b2.String()` equals cannedEnvelopeWithHello ?

**Review by neiljerram — APPROVED** [review]
LGTM, one thought.

File: typha/pkg/syncproto/sync_proto_test.go
Comment (fasaxc) [reply typha/pkg/syncproto/sync_proto_test.go:80]: I deliberately didn't do that; I figured the gob encoding may get inconsequential(?) tweaks over time so it may end up too precise a test.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by fasaxc** [comment]
CI failure looks like a flake

**Pull request merged**
